diff --git a/src/index.js b/src/index.js
index 9b155207d562ce8ac4c14665fada5f978a9a31c4..192c96d7572ef9faab3bd92550a6be754799b602 100644
--- a/src/index.js
+++ b/src/index.js
@@ -71,6 +71,7 @@ export function compile_to_volar_mappings(source, filename, options) {
 		...options,
 		collect: true,
 		loose: !!options?.loose,
+		moduleScopedHookComponents: false,
 		typeOnly: true,
 		errors,
 		comments,
diff --git a/src/transform.js b/src/transform.js
index 7ffccb7f8a49085c41c26b36fa24c312431bf1ef..26249f68f640c160d03cb634488badd3585367d3 100644
--- a/src/transform.js
+++ b/src/transform.js
@@ -45,6 +45,10 @@ const preact_platform = {
 	validation: {
 		requireUseServerForAwait: false,
 	},
+	hooks: {
+		moduleScopedHookComponents: true,
+		componentBodyHookHelpers: true,
+	},
 };
 
 export const transform = createJsxTransform(preact_platform);
